Firestore/core/src/remote/firebase_platform_logging_noop.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_REMOTE_FIREBASE_PLATFORM_LOGGING_NOOP_H_
+#define FIRESTORE_CORE_SRC_REMOTE_FIREBASE_PLATFORM_LOGGING_NOOP_H_
+
+#include <memory>
+#include <string>
+
+#include "Firestore/core/src/remote/firebase_platform_logging.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+class FirebasePlatformLoggingNoOp : public FirebasePlatformLogging {
```

Comment:
Optional: if the base class were to have a default no-op implementation, this class would be unnecessary.

At that point, anyone that wanted an implementation could also just `absl::make_unique<FirebasePlatformLogging>()` for themselves, also removing the need for `CreateNoOpFirebasePlatformLogging`.

Note that I have no strong preference about whether or not there's an explicit no-op type. I can see the case for it but this could also be simpler without it.

Secondarily, I find the disagreement in word ordering between `FirebasePlatformLoggingNoOp` and `CreateNoOpFirebasePlatformLogging` somewhat jarring. I suggest picking just one. Customarily the ordering for refinements of a base type seems to be refinement-base (as in the Create function) so I favor that ordering, but also don't feel strongly. See the differences searching google3 for `\w+ThreadPool\b lang:c++` vs `\bThreadPool\w+ lang:c++`: the former are refinements of thread pool while the latter are mostly either related types (like `ThreadPoolOptions`) or classes where `ThreadPool` is the refinement of some other base type (like `ThreadPoolTaskSystem`).

Reply:
Made the word order consistent.

I guess I have a strong preference towards the `base-refinement` ordering because it creates natural grouping, though there is a good argument to be made regarding precedent. I'd certainly prefer not to tackle it in this PR.

I'm slightly in favor of having an explicit no-op class because it's not supposed to be used normally. I can move it under `test/` in a follow-up if you prefer.